sphinx_gallery/gen_rst.py

```diff
@@ -49,9 +52,9 @@ def prefixed_lines():
         return ''.join(prefixed_lines())
 
 try:
-    from StringIO import StringIO
+    from BytesIO import BytesIO
 except ImportError:
-    from io import StringIO
+    from io import BytesIO
```

Comment:
Yeah, the nastiness of only getting the error when doing `.getvalue()` is what drove me to use `BytesIO` (it doesn't have this problem because you explicitly set the bytes yourself) but I'll change it to `StringIO`